text/0000-unused_const_fn_result_lint.md

```diff
@@ -0,0 +1,173 @@
+- Feature Name: unused_const_fn_result_lint
+- Start Date: 2018-05-29
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a lint for unused results of `const fn` functions,
+if we know for sure that the invocation is dead code.
```

Comment:
IIRC 'dead code' is synonymous with 'unreachable code'. The invocations in your examples aren't unreachable; it's just that their results are discarded (and since they are free of side effects, they can be optimised away).